feat: port all passkey related changes

muXxer opened the pull request:
# Description of change

This PR ports all passkey related upstream changes including refactors and flaky test fixes.

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration tests)
- [ ] Patch-specific tests (correctness, functionality coverage)

### Release Notes

- [x] Protocol: allow passkey in multisig
- [ ] Nodes (Validators and Full nodes):
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

semenov-vladyslav committed:
upstream-changes(zklogin/passkey): use same signing message as other schemes for consistency (#6483)

# Description of change

Port of
https://github.com/MystenLabs/sui/commit/9b79eb3ca038513b01f42b18fa2353db03b6ad98

## Type of change

- Upstream commit port

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any
relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items
that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

### Release Notes

<!--
Check each box that your changes affect. If none of the boxes relate to
your changes, release notes aren't required.

For each box you select, include information after the relevant heading
that describes the impact of your changes that a user might notice and
any actions they must take to implement updates.
-->

- [ ] Protocol:
- [ ] Nodes (Validators and Full nodes):
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

semenov-vladyslav committed:
upstream-changes(zklogin/passkey): Fix a few flaky tests (#6482)

# Description of change

Port of
https://github.com/MystenLabs/sui/commit/0f6c1293b44c77989ee90dbf4d13de97c787ae5b

This fixes a few flaky tests:
1. The multisig tests. If epoch duration is too small, some times it
triggered epoch changes in the middle of test and the max epoch becomes
insufficient.
2. The original upstream commit fixed transaction e2e tesets with
message: "The query includes system transactions too, which there can be
a lot more than 2 pages if enough checkpoints have passed by." The test
fixed was test_get_fullnode_transaction which we don't have.
3. In rosetta with_draw_stake test, similar issue to (1), we need to
manually control epoch change otherwise accidental epoch change could
lead to wrong stake results.

## Links to any relevant issues

## Type of change

- Upstream commit port

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any
relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items
that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

semenov-vladyslav committed:
upstream-changes(zklogin/passkey): allow passkey in multisig (#6487)

Port of
https://github.com/MystenLabs/sui/commit/050f3b9536b07ab55aa7e80aa275020586ff0a11

- Upstream commit port

alexsporn commented on crates/iota-protocol-config/src/lib.rs:
```diff
@@ -45,7 +45,7 @@ pub const MAX_PROTOCOL_VERSION: u64 = 7;
 //            Enable consensus garbage collection for testnet
 //            Enable the new consensus commit rule for testnet.
-
+//            Enable passkey auth in multisig for testnet.
```
This should be added to devnet first

alexsporn commented on crates/iota-protocol-config/src/lib.rs:
```diff
@@ -1961,6 +1969,8 @@ impl ProtocolConfig {
                         // to be included before be considered garbage collected.
                         cfg.consensus_gc_depth = Some(60);
+                        // Enable passkey in multisig.
+                        cfg.feature_flags.accept_passkey_in_multisig = true;
```
This should be added to devnet first

Thoralf-M commented on crates/iota-e2e-tests/tests/multisig_tests.rs:
```diff
@@ -41,6 +56,227 @@ async fn do_upgraded_multisig_test() -> IotaResult {
+    let store: Option<Passkey> = None;
+    let my_authenticator = Authenticator::new(my_aaguid, store, user_validation_method);
+    let mut my_client = Client::new(my_authenticator);
+    let origin = Url::parse("https://www.iota.io").unwrap();
```
```suggestion
    let origin = Url::parse("https://www.iota.org").unwrap();
```

Thoralf-M commented on crates/iota-types/src/unit_tests/passkey_authenticator_test.rs:
Why is this and the next test commented?

Thoralf-M replied:
https://github.com/iotaledger/iota/issues/6991

semenov-vladyslav replied:
Removed tests for now, will re-enable them in a following PR.

muXxer committed:
chore(node): cleanup passkey protocol flags and tests (#7192)

# Description of change

This PR does some cleanups for the passkey feature branch.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

---------

Co-authored-by: Vlad Semenov <vlad.semenov@iota.org>

kodemartin reviewed (approved):
lgtm
